src/main/java/omero/gateway/LoginCredentials.java

```diff
@@ -121,7 +121,18 @@ public LoginCredentials(String username, String password, String host,
         user.setUsername(username);
         user.setPassword(password);
         server.setHostname(host);
-        server.setPort(port);
+        if (port >= 0) {
+            server.setPort(port);
+        }
+        else {
+            // set default ports
+            if (!server.isURL()) {
+                server.setPort(omero.constants.GLACIER2PORT.value);
+            }
+            else if(server.getPort() < 0) {
+                server.setPort(443);
```

Comment:
should we not retrieve that from config?

Reply:
Is there an `omero.constants.WSSPORT` or something like that?

Reply:
No, in python it's hard-coded in clients.py: https://github.com/ome/omero-py/blob/v5.5.dev2/src/omero/clients.py#L333-L339

I guess adding a constant means rebuilding the model?

Reply:
Note the Python client sets the default port to `80` when an unencrypted websockets `ws://` are specified, though we could decide not to offer `ws` for security.